app/presenters/two_factor_auth_code/generic_delivery_mode_presenter.rb

```diff
@@ -0,0 +1,42 @@
+module TwoFactorAuthCode
+  class GenericDeliveryModePresenter
+    include ActionView::Helpers::TagHelper
+    include ActionView::Helpers::TranslationHelper
+    include Rails.application.routes.url_helpers
+
+    def initialize(data_model)
+      data_model.each do |key, value|
+        instance_variable_set("@#{key}", value)
+        self.class.send(:attr_reader, key.to_sym)
```

Comment:
this is a strange pattern to me? dynamically updating the class after an instance has been created?

It may be less flexible, but more explicit IMO if we just `attr_reader` the attributes, doesn't look like there are too many?

```ruby
attr_reader :phone_number, :code_value, :delivery_method, :reenter_phone_number_path, :resend_code_path, :totp_enabled, :unconfirmed_phone
```